query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -125,16 +130,20 @@ public void handleRequest(
     final Context timerContext = pinotQueryExecutionTimer.time();
     try {
       final ResultSetGroup resultSetGroup = pinotClient.executeQuery(pql.getKey(), pql.getValue());
-      timerContext.stop();
 
       if (LOG.isDebugEnabled()) {
         LOG.debug("Query results: [ {} ]", resultSetGroup.toString());
       }
       // need to merge data especially for Pinot. That's why we need to track the map columns
       convert(resultSetGroup, collector, requestAnalyzer.getSelectedColumns());
-      long requestTimeMs = System.currentTimeMillis() - start;
-      if (requestTimeMs > SLOW_REQUEST_THRESHOLD_MS) {
-        LOG.warn("Query Execution time: {} millis\nQuery Request: {}", requestTimeMs, request);
+
+      try {
+        long requestTimeMs = System.currentTimeMillis() - start;
+        if (requestTimeMs > SLOW_REQUEST_THRESHOLD_MS) {
```

Comment:
Can we make this threshold configurable? for debugging purpose, in some use cases 3s may not be acceptable threshold. it's okay to put in different PR though, since it is outside the scope of this review.

Reply:
Yeah, thought about it. The current way code is structured, it's not straightforward so deferred it but I'm planning to address in future PRs when I get time.